iGCS/Managers/GCSDataManager.m

```diff
@@ -20,13 +23,46 @@ + (instancetype) sharedInstance {
     return sharedInstance;
 }
 
+
 - (instancetype) init {
     self = [super init];
     if (self) {
         _craft = [GCSCraftModelGenerator createInitialModel];
         _lastViewedMapCamera = nil;
+       [self loadArchives];
     }
     return self;
 }
 
+- (void)loadArchives {
+    NSData *decodedData = [NSData dataWithContentsOfFile:[GCSDataManager filePath]];
+    GCSSettings *progData = [NSKeyedUnarchiver unarchiveObjectWithData:decodedData];
+    if (progData) {
+        _gcsSettings = progData;
+    }
+    else {
+        _gcsSettings = [[GCSSettings alloc] init];
+    }
+}
+
+
++ (void) save {
+    NSData* encodeData = [NSKeyedArchiver archivedDataWithRootObject:[GCSDataManager sharedInstance].gcsSettings];
+    [encodeData writeToFile:[GCSDataManager filePath] atomically:YES];
+}
+
++(NSString*)filePath
+{
+    static NSString* filePath = nil;
+    if (!filePath) {
+        filePath =
+        [[NSSearchPathForDirectoriesInDomains(NSDocumentDirectory, NSUserDomainMask, YES) firstObject]
```

Comment:
Should be able to use `[[FileUtils URLToFileInDocumentsDirWithFileName:logName] path]` instead of this